Source/items.cpp

```diff
@@ -3221,6 +3205,7 @@ static bool OilItem(ItemStruct *x, PlayerStruct *p)
 		break;
 	case IMISC_OILBSMTH:
 		if (x->_iMaxDur != 255) {
+			int dur;

```

Comment:
Avoid uninitialized values
```suggestion
			int dur = x->_iMaxDur + 1;
```
(a few of the lines below can then use `dur` instead of `x->_iMaxDur`)

Reply:
That would break the logical flow of the code, even though the end result is still the same. I refactored it even further.